presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -191,6 +194,46 @@ public long zOrderByteAddressToLong(byte[] byteAddress)
         return byteAddress;
     }
 
+    /**
+     * Searches for and outputs ranges of long addresses within certain ranges in each dimension.
+     */
+    public List<ZAddressRange> zOrderCondensedSearchCurveLongs(List<ZValueRange> ranges)
+            throws Exception
+    {
+        curveSearcher.setReturnLong(true);
```

Comment:
Instead of using setter, have we considered making this a enum, and passed to `zOrderCondensedSearchCurve`? Currently `returnLong = false` means `int`, which might not be intuitive.